src/main/java/com/example/training/configuration/SecurityInterceptor.java

```diff
@@ -15,12 +17,13 @@
 @Component
 public class SecurityInterceptor implements HandlerInterceptor {
 
-    private final PoliciesService policiesService;
-    private final AccidentsService accidentsService;
+    private final ObjectProvider<PoliciesService> policiesServiceProvider;
```

Comment:
I had several problems when I changed the Feign client to use it as a bean. This was the only thing that worked for me and I don't really understand it. The problem was related with cyclic references with dependencies. I'd like to take a look at this carefully.

Reply:
This shouldn't be necessary, and I'm not seeing what's happening to you but we should check this out and fix it.